ghcide/src/Development/IDE/Import/FindImports.hs

```diff
@@ -147,13 +148,26 @@ locateModule env comp_info exts targetFor modName mbPkgName isSource = do
             map snd import_paths
 #endif
 
-      mbFile <- locateModuleFile ((homeUnitId_ dflags, importPaths dflags) : import_paths') exts targetFor isSource $ unLoc modName
+      mbFile <- locateModuleFile ((homeUnitId_ dflags, importPaths dflags) : other_imports) exts targetFor isSource $ unLoc modName
       case mbFile of
         Nothing          -> lookupInPackageDB env
         Just (uid, file) -> toModLocation uid file
   where
     dflags = hsc_dflags env
     import_paths = mapMaybe (mkImportDirs env) comp_info
+    other_imports =
+#if MIN_VERSION_ghc(9,4,0)
+      -- On 9.4+ instead of bringing all the units into scope, only bring into scope the units
+      -- this one depends on
+      map (\uid -> (uid, importPaths (homeUnitEnv_dflags (ue_findHomeUnitEnv uid ue)))) hpt_deps
+    ue = hsc_unit_env env
+    units = homeUnitEnv_units $ ue_findHomeUnitEnv (homeUnitId_ dflags) ue
```

Comment:
This will throw if the unit id can't be found, right? Not much we can do, though.

Reply:
the home unit id should always be set in ghcide.

Reply:
Yeah, I agree